stairctl/src/main/java/bio/terra/stairctl/commands/Output.java

```diff
@@ -0,0 +1,222 @@
+    } else {
+      System.out.println(String.format(STAIRWAY_LIST_FORMAT, "\nStairwayId"));
+      System.out.println(STAIRWAY_LIST_DASH);
+      for (String id : stairwayList) {
+        System.out.println(String.format(STAIRWAY_LIST_FORMAT, id));
+      }
+      System.out.println();
+    }
+  }
+
+  // Simple case with no input map
+  public static void flightSummary(Control.Flight flight) {
+    flightSummary(flight, null);
+  }
+
+  public static void flightSummary(Control.Flight flight, List<Control.KeyValue> inputMap) {
+    final String indent = "  ";
+    List<ImmutablePair<String, String>> pairList = new ArrayList<>();
+    pairList.add(new ImmutablePair<>("class", flight.getClassName()));
+    pairList.add(new ImmutablePair<>("submitted", flight.getSubmitted().toString()));
+    pairList.add(
+        new ImmutablePair<>(
+            "completed", flight.getCompleted().map(Instant::toString).orElse(StringUtils.EMPTY)));
+    pairList.add(new ImmutablePair<>("status", flight.getStatus().toString()));
+    pairList.add(new ImmutablePair<>("exception", flight.getException().orElse(StringUtils.EMPTY)));
+    pairList.add(
+        new ImmutablePair<>("stairwayId", flight.getStairwayId().orElse(StringUtils.EMPTY)));
+
+    System.out.println(String.format("%nFlight: %s", flight.getFlightId()));
+    int maxLength = display(indent, pairList);
+    if (inputMap != null) {
+      String inputTitle = "%s%-" + maxLength + "s:";
+      System.out.println(String.format(inputTitle, indent, "inputMap"));
+      keyValue(indent + "  ", inputMap);
+    }
+  }
+
+  public static void showConnection(ConnectParams connectParams) {
+    List<ImmutablePair<String, String>> pairList = new ArrayList<>();
+    pairList.add(new ImmutablePair<>("dbname", connectParams.getDbname()));
+    pairList.add(new ImmutablePair<>("username", connectParams.getUsername()));
+    pairList.add(new ImmutablePair<>("host", connectParams.getHost()));
+    pairList.add(new ImmutablePair<>("port", connectParams.getPort()));
+    System.out.println("Stairway Connection:");
+    display("  ", pairList);
+  }
+
+  private static int display(String indent, List<ImmutablePair<String, String>> pairList) {
+    // Compute the max length of the first string
+    Optional<String> longest =
+        pairList.stream().map(Pair::getLeft).max(Comparator.comparing(String::length));
+    if (longest.isEmpty()) {
+      return 0;
+    }
+    int maxLength = longest.get().length();
+    String displayFormat = "%s%-" + maxLength + "s: %s";
+    for (ImmutablePair<String, String> pair : pairList) {
+      System.out.println(String.format(displayFormat, indent, pair.getLeft(), pair.getRight()));
+    }
+    return maxLength;
+  }
+
+  public static void error(String userMessage, Exception ex) {
+    System.err.println(": " + ex.getMessage());
+    logger.error(userMessage + ": ", ex);
+  }
+
+  private static String shortenClassName(String in) {
+    if (in.length() <= 30) {
```

Comment:
I think the class name includes the package. Instead of truncating the end of the class name, what if we removed or abbreviated the package name first? Maybe overkill.

Reply:
I found an Apache commons method to do the abbreviation, so it will be like what we see in the logs.